tests/FactoryTest.php

```diff
@@ -192,6 +193,31 @@ public function it_lets_you_use_faker_for_defining_data(): void
         $this->assertIsString($group->mobile);
     }
 
+    /** @test * */
+    public function it_lets_you_use_a_closure_for_defining_default_data(): void
+    {
+        $ingredient = IngredientFactoryUsingClosure::new()->create();
+
+        $this->assertIsString($ingredient->name);
+        $this->assertIsString($ingredient->description);
+        $this->assertEquals($ingredient->name, 'Pasta');
+        $this->assertEquals($ingredient->description, 'Super delicious Pasta');
+    }
+
+    /** @test * */
+    public function it_lets_you_use_a_closure_for_overriding_default_data(): void
+    {
+        $ingredient = IngredientFactoryUsingClosure::new()->create([
+            'name' => fn (array $ingredient) => 'Basil',
```

Comment:
I updated the test to define two closures. This will accurately test that the second closure can use the value from the first closure.